pkg/store/bucket.go

```diff
@@ -225,10 +226,18 @@ func NewBucketStore(
 		return nil, errors.Errorf("max concurrency value cannot be lower than 0 (got %v)", maxConcurrent)
 	}
 
-	indexCache, err := newIndexCache(reg, indexCacheSizeBytes)
+	// TODO(bwplotka): Add as a flag?
+	maxItemSizeBytes := uint64(float64(indexCacheSizeBytes) * 0.5)
```

Comment:
Assumption: Max item size should be not bigget than 50% of Overall size. I think it's generous already.